### DISCUSSION TITLE
must-see Guido van Rossum interview

### DISCUSSION THREAD

**Discussion by jqnatividad** [ckan/ckan#7250] [General]
A little off-topic, but still related as CKAN is tied at the hip to Python.

It's a 3 hour 22 minute interview with Guido (Python's BDFL - Benevolent Dictator for Life), but it's well worth it! Things discussed:

* Python's beginnings
* Guido's current IDE (VS Code, which he calls his new Emacs)
* Python 4
* GIL
* Why Python is the lingua franca of Machine Learning/Data Science
* and as this is a Lex Fridman podcast, a sprinkling of philosophy here and there 😉

https://lexfridman.com/guido-van-rossum-2/

**Comment by thegostev** [comment]
Thanks for suggesting @jqnatividad I did enjoy this interview. Discussion over why Python is popular amongst data scientists made me think about extrapolating this question to CKAN - how it became so popular for open data?

**Reply by jqnatividad** [reply]
Glad you enjoyed it @thegostev!

IMHO, CKAN's popularity is primarily because its an extensible data management framework, it powers the largest open data portals (which has become showcase implementations), and its open source architecture has enabled an organic, vibrant ecosystem of contributors.

I actually think that there is unrealized potential with CKAN for the very same data scientists to embrace it as their de facto Data Management System.

Where it becomes the single source of truth for metadata about their disparate data assets - not just conventional metadata, but stuff like their provenance, how they're produced, how much they cost, etc.

Where the very same data pipelines that produce these assets - get, create and maintain the very same metadata when they're executed (e.g. which version of the incoming data should it use before it starts, and where it updates the metadata of the output it produces). 

Where they can, not only catalogue these assets, but also catalogue and tell stories about their analyses and data products they've produced - their Jupyter notebooks, their plot.ly visualizations, etc.

Where they register their enriched data (which in turn, becomes the input of another pipeline).

My hope is that with our [harmonized research effort](https://ckan.org/events/ckan-monthly-live-november-ckan-pathways-to-3-0), we can help CKAN realize its full potential.
